ev3dev2/motor.py

```diff
@@ -2099,6 +2111,169 @@ def follow_line(self,
 
         self.stop()
 
+    def calibrate_gyro(self):
+        """
+        Calibrates the gyro sensor.
+
+        NOTE: This takes 1sec to run
+        """
+        assert self.gyro, "GyroSensor must be defined"
+
+        for x in range(2):
+            self.gyro.mode = 'GYRO-RATE'
+            self.gyro.mode = 'GYRO-ANG'
+            time.sleep(0.5)
+
+    def follow_gyro_angle(self,
+            kp, ki, kd,
+            speed,
+            target_angle=0,
+            sleep_time=0.01,
+            follow_for=follow_for_forever,
+            **kwargs
+        ):
+        """
+        PID gyro angle follower
+
+        ``kp``, ``ki``, and ``kd`` are the PID constants.
+
+        ``speed`` is the desired speed of the midpoint of the robot
+
+        ``target_angle`` is the angle we want drive
+
+        ``sleep_time`` is how many seconds we sleep on each pass through
+            the loop.  This is to give the robot a chance to react
+            to the new motor settings. This should be something small such
+            as 0.01 (10ms).
+
+        ``follow_for`` is called to determine if we should keep following the
+            line or stop.  This function will be passed ``self`` (the current
+            ``MoveWithGyroTank`` object). Current supported options are:
+            - ``follow_for_forever``
+            - ``follow_for_ms``
+
+        ``**kwargs`` will be passed to the ``follow_for`` function
+
+        Example:
+
+        .. code:: python
+
+            from ev3dev2.motor import OUTPUT_A, OUTPUT_B, MoveTank, SpeedPercent, follow_for_ms
+            from ev3dev2.sensor.lego import GyroSensor
+            tank = MoveTank(OUTPUT_A, OUTPUT_B)
+            tank.gyro = GyroSensor()
+            try:
+                # Follow the line for 4500ms
+                tank.calibrate_gyro()
+                tank.follow_gyro_angle(
```

Comment:
Where is the target angle?

What's the expected use case for this? Is the idea that you'd first retrieve the current angle then follow that? I'm in particular wondering how you know what target angle to use.

Reply:
Typo in my example code. Good catch!